SmvInterceptorWrapper/Program.cs

```diff
@@ -118,29 +134,81 @@ static int Main(string[] args)
+
                 ProcessStartInfo psi = new ProcessStartInfo(System.IO.Path.GetFullPath(Environment.ExpandEnvironmentVariables("%SMV_ANALYSIS_COMPILER%")), Environment.ExpandEnvironmentVariables(rspContents));
                 psi.RedirectStandardError = true;
                 psi.RedirectStandardOutput = true;
                 psi.UseShellExecute = false;
-
-                //Console.WriteLine("starting " + psi.ToString());
-
+                
                 if (!psi.EnvironmentVariables.ContainsKey("esp.cfgpersist.persistfile"))
                 {
                     psi.EnvironmentVariables.Add("esp.cfgpersist.persistfile", smvOutDir + "\\$SOURCEFILE.rawcfgf");
                     psi.EnvironmentVariables.Add("Esp.CfgPersist.ExpandLocalStaticInitializer", "1");
                     psi.EnvironmentVariables.Add("ESP.BplFilesDir", smvOutDir);
                 }
                 
-                //Console.WriteLine("iwrap: cl.exe --> " + psi.FileName + " " + psi.Arguments);
+                WriteInterceptorLog("LAUNCH: iwrap: " + psi.FileName + " " + psi.Arguments);
+                WriteInterceptorLog("PATH: " + Environment.ExpandEnvironmentVariables("%PATH%"));
+                
+                string environmentVars = "";
+                environmentVars += "esp.cfgpersist.persistfile=";
+                environmentVars += (smvOutDir + "\\$SOURCEFILE.rawcfgf");
+                environmentVars += "  ";
+                environmentVars += "Esp.CfgPersist.ExpandLocalStaticInitializer=1";
+                environmentVars += "  ";
+                environmentVars += "ESP.BplFilesDir=";
+                environmentVars += smvOutDir;
+                WriteInterceptorLog("Process-specific environment variables: " + environmentVars);
 
                 Process p = System.Diagnostics.Process.Start(psi);
 
                 smvclLogContents.Append(p.StandardOutput.ReadToEnd());
                 smvclLogContents.Append(p.StandardError.ReadToEnd());
                 File.WriteAllText(smvclLogPath, smvclLogContents.ToString());
                 p.WaitForExit();
+                
+                WriteInterceptorLog("EXIT: CL.exe.  Exit code: " + p.ExitCode);
+
+                if (debugMode)
+                {
+                    // Run with /P to get preprocessed output for debugging
+                    psi = new ProcessStartInfo(System.IO.Path.GetFullPath(Environment.ExpandEnvironmentVariables("%SMV_ANALYSIS_COMPILER%")), Environment.ExpandEnvironmentVariables(rspContentsDebug));
+                    psi.RedirectStandardError = true;
+                    psi.RedirectStandardOutput = true;
+                    psi.UseShellExecute = false;
+                    p = System.Diagnostics.Process.Start(psi);
+
+                    // Allow 5 minutes for this, then continue; don't hang indefinitely on debug
+                    p.WaitForExit(5 * 60 * 1000);
```

Comment:
can we define this as a constant in a config file. easier to change for debugging etc.

Reply:
Moved it to SMV_DEBUG_TIMEOUT environment variable, which can be added to InterceptedBuild XML, etc